code/datums/outputs.dm

```diff
@@ -0,0 +1,86 @@
+GLOBAL_LIST_EMPTY(outputs_list)
+
+/datum/outputs
+	var/text = ""
+	var/list/sounds = list('sound/items/airhorn.ogg'=1) //weighted, put multiple for random selection between sounds
+	var/list/icon = list('icons/sound_icon.dmi',"circle", HUD_LAYER) //syntax: icon, icon_state, layer
+
+/datum/outputs/New()
+	GLOB.outputs_list[type] = src
+
+/datum/outputs/proc/send_info(mob/receiver, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	var/sound/S
+	if(receiver.client)
+		//Pick sound
+		if(sounds.len)
+			var/soundin = pickweight(sounds)
+			S = sound(get_sfx(soundin))
+		//Process sound
+		if(S && receiver.can_hear())
```

Comment:
You dont need to to check if reciever can hear, reciever will check its hearing ability itself